### ISSUE TITLE
Errors running entrypoint-related commands

### ISSUE THREAD

**Issue by paladin-devops** [hashicorp/waypoint#2556]
**Describe the bug**
Commands such as waypoint logs and waypoint exec do not yield expected results

**Steps to Reproduce**
Build container with Docker builder plugin. Deploy artifact to Nomad via Nomad jobspec plugin. 

waypoint.hcl
```
project = "my-project"

app "my-app" {
  build {
    use "docker" {
      dockerfile = templatefile("${path.app}/Dockerfile", {
        tag = var.tag
      })
    }
    registry {
      use "docker" {
        image        = "my-repo/${app.name}"
        tag          = var.tag
        encoded_auth = var.auth
      }
    }
  }

  deploy {
    use "nomad-jobspec" {
      jobspec = templatefile("${path.app}/${app.name}.waypoint.nomad.hcl", {
        env         = workspace.name
        password = var.password
      })
    }
  }
}

//vars declared
```

waypoint logs output:
```
waypoint logs
! Error reading logs: rpc error: code = Unavailable desc = error reading from server: EOF
```

waypoint exec /bin/sh output:
```
2021-10-21T19:01:37.995-0400 [INFO]  waypoint: waypoint version: full_string="v0.6.1 (1b85b054)" version=v0.6.1 prerelease="" metadata="" revision=1b85b054
2021-10-21T19:01:37.997-0400 [TRACE] waypoint: starting interrupt listener for context cancellation
2021-10-21T19:01:37.997-0400 [TRACE] waypoint: interrupt listener goroutine started
2021-10-21T19:01:37.998-0400 [DEBUG] waypoint: home configuration directory: path=/Users/user/Library/Preferences/waypoint
2021-10-21T19:01:38.004-0400 [TRACE] waypoint: no API client provided, initializing connection if possible
2021-10-21T19:01:38.004-0400 [TRACE] waypoint.server: WithLocal set, server credentials optional
2021-10-21T19:01:38.004-0400 [INFO]  waypoint.server: attempting to source credentials and connect
2021-10-21T19:01:38.005-0400 [DEBUG] waypoint.serverclient: connection information: address=<REDACTED> tls=true tls_skip_verify=true send_auth=true has_token=true
2021-10-21T19:01:38.182-0400 [DEBUG] waypoint.server: connection established with sourced credentials
2021-10-21T19:01:38.182-0400 [TRACE] waypoint: requesting version info from server
2021-10-21T19:01:38.233-0400 [INFO]  waypoint: server version info: version=v0.6.1 api_min=1 api_current=1 entrypoint_min=1 entrypoint_current=1
2021-10-21T19:01:38.233-0400 [INFO]  waypoint: negotiated api version: version=1
2021-10-21T19:01:38.233-0400 [DEBUG] waypoint: starting runner to process local jobs
2021-10-21T19:01:38.233-0400 [DEBUG] waypoint.runner: Created runner: id=01FJJHDA9S1EZMTMXY5SJ42D3R
2021-10-21T19:01:38.233-0400 [DEBUG] waypoint.runner: registering runner
2021-10-21T19:01:38.233-0400 [DEBUG] waypoint.runner: runner registered, waiting for first config processing
2021-10-21T19:01:38.289-0400 [INFO]  waypoint.runner.config_recv: new configuration received
2021-10-21T19:01:38.289-0400 [TRACE] waypoint.runner.watch_config.watchloop: got source config update but ignoring since there is no diff
2021-10-21T19:01:38.289-0400 [DEBUG] waypoint.runner.watch_config.watchloop: new config variables received, scheduling refresh
2021-10-21T19:01:38.289-0400 [TRACE] waypoint.runner.watch_config.watchloop: calculating changes between old and new config
2021-10-21T19:01:38.791-0400 [TRACE] waypoint.runner.watch_config.watchloop: refreshing app configuration
2021-10-21T19:01:38.792-0400 [DEBUG] waypoint.runner.watch_config.watchloop: new configuration computed
2021-10-21T19:01:38.792-0400 [TRACE] waypoint.runner.watch_config: received new app config
2021-10-21T19:01:38.792-0400 [INFO]  waypoint.runner: runner registered with server and ready
2021-10-21T19:01:38.792-0400 [DEBUG] waypoint: will operate on app: name=my-app
2021-10-21T19:01:38.792-0400 [DEBUG] waypoint.runner: opening job stream
2021-10-21T19:01:38.792-0400 [DEBUG] waypoint: looking up deployments to use for app: app="application:"my-app"  project:"my-project""
2021-10-21T19:01:38.792-0400 [TRACE] waypoint.runner: sending job request
2021-10-21T19:01:38.792-0400 [INFO]  waypoint.runner: waiting for job assignment
2021-10-21T19:01:38.841-0400 [DEBUG] waypoint: found deployment: deployment-id=01FJJGW983BX6G7GZMN50DR64J
⠙ Waiting for instance assignment...2021-10-21T19:01:38.889-0400 [WARN]  waypoint.runner.watch_config: exiting due to context ended
2021-10-21T19:01:38.889-0400 [TRACE] waypoint.runner.config_recv: exiting receive goroutine
2021-10-21T19:01:38.889-0400 [TRACE] waypoint.runner.watch_config: exiting goroutine
! No running instances found for exec!
  
  If you just recently deployed, the instances could still be starting up.
  Otherwise, please diagnose the issue by inspecting your application logs.
  If application logs are not available, the application may have failed to start.
2021-10-21T19:01:38.889-0400 [TRACE] waypoint: stopping signal listeners and cancelling the context
```

Nomad jobspec excerpt (showing injection of entrypoint environment variables to Nomad job)
```
      env {
        %{ for k,v in entrypoint.env ~}
        ${k} = "${v}"
        %{ endfor ~}
        WAYPOINT_LOG_LEVEL = "DEBUG"
      }
```

Actual env vars from Nomad job definition:
![Screen Shot 2021-10-21 at 7 06 43 PM](https://user-images.githubusercontent.com/83741749/138368771-f3d531e7-6ef6-49d4-a8c8-06d6519504bd.png)

**Expected behavior**
Logs written to console, successful exec into running container.

**Waypoint Platform Versions**
CLI: 0.6.1
Server: 0.6.1
Plugins: Docker builder, Nomad jobspec deployer

**Comment by evanphx** [comment]
But did you use `waypoint install` on this new server? That is the phase that tells waypoint server about itself (it's address) that allows entrypoints to connect back to it.

**Comment by paladin-devops** [comment]
@evanphx since your last comment, I've re-deployed my server after 0.6.2 was released. I've made no customizations after running waypoint install which would have led to the entrypoints being confused about how to connect back to the server. And on one of my apps, the Waypoint Entrypoint is actually working now (I was able to exec and display logs)! However even with the same configurations on another app (entrypoint is enabled in the build, a nd the entrypoint env vars are supplied in the deploy), the second app is not connecting back. I suspect it's because I hadpreviously built the 2nd app and disabled the entrypoint in the build phase, but then re-enabled it after.
